File: SuperBAS.Transpiler.Configs/CSharp/skeleton.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Threading;
using System.IO;

namespace UserProgram
{
    class Program
    {
        private static Random rand;
        /*DECLARATIONS*/

        static void Gosub(double lineNumber)
        {
        GosubStart:
            switch (lineNumber)
            {
                case -1:
                    return;
                /*BODY*/
                default:
                    throw new Exception($"Invalid GOTO { lineNumber } - Not a line");
            }
        }

        static void Main(string[] args)
        {
            Gosub(/*LOWESTLINE*/);
        }

        static string ReadAllFile (string flName) {
            // Read an entire file to a string
            // Put into a function for use in WRITEFILE command
            var sr = new StreamReader(flName);
            var s = sr.ReadToEnd();
            sr.Close();
            return s;
        }
    }
}

File: SuperBAS.Transpiler.Configs/JavaScript/skeleton.js
/*DECLARATIONS*/
let stop = false
const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms))

async function goSub (line) {
  switch (line) {
    case -1:
      return
    /*BODY*/
    break
    default:
      throw new Error(`Can't GOTO ${line} - invalid line number.`)
  }
}

function readAllFile () {
  console.log('READALLFILE is not yet supported in JS.')
}

function writeAllFile (flName, contents, append) {
  console.log('WRITEALLFILE is not yet supported in JS.')
}

function newDim (...dimensions) {

}
function getArrayValue (array, ...dimensions) {

}

goSub(/*LOWESTLINE*/)
